code/_onclick/hud/screen_objects.dm

```diff
@@ -617,3 +617,121 @@
 
 	user.hud_used.hidden_inventory_update()
 	return TRUE
+
+/obj/screen/ammo
+	name = "ammo"
+	icon = 'icons/mob/ammoHUD.dmi'
+	icon_state = "ammo"
+	screen_loc = ui_ammo
+	var/warned = FALSE
+
+/obj/screen/ammo/proc/add_hud(var/mob/user)
+	if(!user?.client)
+		return
+
+	if(user.client.screen.Find(src))
+		return
+	else
+		user.client.screen += src
+
+/obj/screen/ammo/proc/remove_hud(var/mob/user)
+	if(!user?.client)
+		return
+
+	if(!user.client.screen.Find(src))
+		return
+	else
+		user.client.screen -= src
+
+/obj/screen/ammo/proc/update_hud(var/mob/user)
+	if(!user?.client || !user.client.screen.Find(src))
+		return
+
+	var/obj/item/weapon/gun/G = user.get_active_hand()
+
+	if(!G)
+		return
+
+	var/hud_state
+	var/hud_state_empty
+
+	var/rounds = 0
+
+	if(G.current_mag)
+		rounds += G.current_mag.current_rounds
+
+	if(G.in_chamber && !istype(G, /obj/item/weapon/gun/launcher) && !istype(G, /obj/item/weapon/gun/revolver) && !istype(G, /obj/item/weapon/gun/smartgun) && !istype(G, /obj/item/weapon/gun/energy))
+		rounds++
+
+	if(istype(G, /obj/item/weapon/gun/energy))
+		var/obj/item/weapon/gun/energy/E = G
+		rounds += (E.cell.charge / E.charge_cost)
```

Comment:
Do an
```
else if(G.in_chamber && !istype(G, /obj/item/weapon/gun/launcher) && !istype(G, /obj/item/weapon/gun/revolver) && !istype(G, /obj/item/weapon/gun/smartgun))
	rounds++
```
here, avoiding the double-check.
Or even better, add a variable in those guns to account for this, instead of checking every type.

Reply:
Also a safe check for a division-by-zero would be good.